### PR TITLE
Rename accession fields (DataBiosphere/azul#3331)

### LINKED ISSUE #3331: Add dbGAP, BioStudies and EGA accessions to project response
Add `project.dbgap_accessions`, `project.ega_accessions` and `project.biostudies_accessions` to set of returned accessions.

~Solve #3331, #3332 and #3333 in the same PR.~

 Modify `hits.projects ` in the service response from

```json
{
    …
    "arrayExpressAccessions": [
        null
    ],
    "geoSeriesAccessions": [
        "GSE93421"
    ],
    "insdcProjectAccessions": [
        "SRP096558"
    ],
    "insdcStudyAccessions": [
        "PRJNA360949"
    ]
}
```

to 

```json
{
    …
    "arrayExpressAccessions": [
        null
    ],
    "geoSeriesAccessions": [
        "GSE93421"
    ],
    "insdcProjectAccessions": [
        "SRP096558"
    ],
    "insdcStudyAccessions": [
        "PRJNA360949"
    ],
    "accessions": [
        {
            "type": "geo_series",
            "value": "GSE93421"
        },
        {
            "type": "insdc_project",
            "value": "SRP096558"
        },
        {
            "type": "insdc_study",
            "value": "PRJNA360949"
        },
        {
            "type": "dbgap",
            "value": "…"
        },
        {
            "type": "ega",
            "value": "…"
        },
        {
            "type": "biostudies",
            "value": "…"
        }
    ]
}
```

The `type` value is the name of the corresponding schema property minus the `_accessions` suffix.

Note that there is no `null` entry for ArrayExpress in `accessions`.

The new `dbgap`, `ega` and `biostudies` type accessions are only exposed in the new `accessions` property. Any other type is exposed in the `accessions` property AND a separate backwards-compatible property.

### PR DESCRIPTION
**Pull request by amarjandu**
From comment : https://github.com/DataBiosphere/azul/pull/3369#discussion_r713198007
See thread: https://humancellatlas.slack.com/archives/C01360XN04S/p1630428305017100

### CODE REVIEW

**Comment by amarjandu** [comment]
Self Note: unable to link PR to issue, looks like its permissions related. Keeping assignment to me as well to track ticket.

File: src/humancellatlas/data/metadata/api.py [hunk 1]
```diff
@@ -318,11 +318,11 @@ def __init__(self,
             if suffix == 'accessions':
                 assert prefix, name
                 assert isinstance(value, list)
-                accessions.update(Accession(domain=prefix, value=v) for v in value)
+                accessions.update(Accession(namespace=prefix, accession=v) for v in value)
         self.accessions = accessions
 
     def _accessions(self, domain: str) -> Set[str]:
-        return {a.value for a in self.accessions if a.domain == domain}
+        return {a.accession for a in self.accessions if a.namespace == domain}
```
Comment (dsotirho-ucsc) [inline src/humancellatlas/data/metadata/api.py]:
```suggestion
        return {a.accession for a in self.accessions if a.namespace == namespace}
```
No reason to keep the parameter named `domain`, right?

**Review by dsotirho-ucsc — CHANGES_REQUESTED** [review]

File: src/humancellatlas/data/metadata/api.py (hunk 1 above)
Comment (amarjandu) [reply src/humancellatlas/data/metadata/api.py]: Thanks for catching this :)

**Review by dsotirho-ucsc — APPROVED** [review]
Approved.

**Comment by amarjandu** [comment]
Pulling this, need to change the linked issue to #3331

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
@jessebrennan this can be merged and released.

**Comment by hannes-ucsc** [comment]
The connected ticket should not be moved to `Merged` after that, because there is still a downstream PR in the works.

**Pull request closed without merging**